doc/release-guide.md

```diff
@@ -0,0 +1,74 @@
+# Releases & Deployments
```

Comment:
@dorner @cielf @scooter-dangle I'd like an early review of this documentation. I suspect that we should be running through this together coming up this Sunday and see if anything is missing.